File: kotlin/math/main_3036.kt
import kotlin.math.min

fun main() {
    val count = readLine()!!.toInt()
    val ring = readLine()!!.split(' ').map { it.toInt() }
    for (i in 1 until count){
        var std = ring[0]
        var cmp = ring[i]
        var j = 2
        while (j <= min(std, cmp)){
            if (std % j == 0 && cmp % j == 0){
                std /= j
                cmp /= j
            } else j++
        }
        println("$std/$cmp")
    }
}

File: cpp/math/main_3036.cpp
#include <iostream>
#include <algorithm>
#include <set>
#include <cmath>
using namespace std;

int gcd(int a, int b) {
	const int cmp = a % b;
	if (cmp == 0) {
		return b;
	}
	return gcd(b, cmp);
}

int main() {
	int count;
	cin >> count;
	int first;
	cin >> first;
	int cmp;
	int g;
	for (int i = 1; i < count; ++i) {
		cin >> cmp;
		g = gcd(max(cmp, first), min(cmp, first));
		cout << first / g << '/' << cmp / g << '\n';
	}
}